### PR TITLE
Card Browser: Fix "Undo" not appearing and stale data

### LINKED ISSUE #7674: Browser - Reposition: does not update cards or add undo
###### Reproduction Steps

1. Select a card
2. Reposition: 1

###### Expected

3. Undo reposition is visible
4. Card has the "due" updated

###### Actual

3. "Undo reposition" is not visible
4. Card has the same "due"

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description

* Pressing "Reschedule/Reposition/Reset Cards" did not bring up the "undo" item
* These items and Undo did not update the data in the grid

## Fixes
Fixes #7674 

## Approach
Obtain the cards, reset them and make the grid refresh

## How Has This Been Tested?

Android 9, and some unit tests - some were flaky

## Learning (optional, can help others)
We need more tests

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] Strings resources: All strings added as resources are unique or contain a comment for seemingly duplicate strings to explain the difference between both resources

### CODE REVIEW

**Comment by david-allison-1** [comment]
## Please note: this conflicts with #7676 - needs rebase after commit

**Review by Arthur-Milchior — APPROVED** [review]

**Comment by Arthur-Milchior** [comment]
I was trying to approve a single commit, I've not yet read the whole PR

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 1]
```diff
@@ -2745,7 +2770,7 @@ void rerenderAllCards() {
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
-    void checkCardsAtPositions(int[] positions) {
+    void checkCardsAtPositions(int... positions) {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:2788]: Nice.
May I suggest to move it in its own commit, and update the four tests where it's used to remove the array and make code smaller

**Comment by Arthur-Milchior** [comment]
I just restarted the two failing tests, since it seems unrelated to this PR.
If it still fails, I'd suggest splitting this PR in two. The NFs change are nice and can be merged quickly, while the actual correction may takes more time.
I'm trying to test code, but I still have the problem "Unable to determine application id: com.android.tools.idea.run.ApkProvisionException: Error loading build artifacts from: /home/milchior/anki/ankidroid/AnkiDroid/build/outputs/apk/debug/output-metadata.json"

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 2]
```diff
@@ -1169,10 +1169,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
                     getString(R.string.reposition_card_dialog_title),
                     getString(R.string.reposition_card_dialog_message),
                     5);
-            repositionDialog.setCallbackRunnable(days ->
-                    CollectionTask.launchCollectionTask(DISMISS_MULTI, repositionCardHandler(),
-                            new TaskData(new Object[] {cardIds, Collection.DismissType.REPOSITION_CARDS, days}))
-            );
+            repositionDialog.setCallbackRunnable(position -> repositionCardsNoValidation(cardIds, position));
```
[the remark this answers is not in the document]
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:1166]: Because `days` was an incorrect variable name

**Review by Arthur-Milchior — APPROVED** [review]

**Comment by Arthur-Milchior** [comment]
> Merge attempt failed
> The selected merge method (merge) is not allowed.

I don't know why. I was trying to merge it

**Review by mikehardy — APPROVED** [review]
Oh wow - has this been broken forever? This is more of my old code, the ability to reset the scheduling on more than a card at once was one of the most requested features way back when and was the first user visible thing I tried! Nice fix

**Comment by mikehardy** [comment]
picked and pushed to release-2.14, not released yet

**Comment by Arthur-Milchior** [comment]
As far as I understand it, reset was correctly done. The problem was refreshing data. And as I wrote on https://github.com/ankidroid/Anki-Android/issues/7674#issuecomment-727173078 I was able to reproduce it with 2.13.5, early october. I did not check earlier, since I just wanted to check whether reverting https://github.com/ankidroid/Anki-Android/pull/7677 would solve the problem and saw that the problem already appeared before https://github.com/ankidroid/Anki-Android/pull/7677 was merged ( but this was reverted nonetheless, probably because of new cards that did not appear in the search result)
